### ISSUE TITLE
Escaping special characters in RegEx breaks it

### ISSUE THREAD

**Issue by supaiku-o** [godotengine/godot#5827]
**Operating system or device - Godot version:**
Arch Linux - c83aa5f86383bf89871f0a41752f5c6eeb148950

**Issue description**
https://godotengine.org/qa/6247/regex-string-literal
Basically, I can't use regex because of special characters. Escaping them renders the regex useless.

**Steps to reproduce:**
```gdscript
var text = "test123"
var regex = RegEx.new()
regex.compile("/\\d+/g")
regex.find(text)
print(text.get_captures()) # => []
```
**What I expected**
https://regex101.com/r/kF6hC9/1

**Comment by akien-mga** [comment]
Ping @leezh who wrote the current library.

**Comment by leezh** [comment]
The compile function expects just the pattern, without the enclosing `//g`. The fix for it is:

    regex.compile("\\d+")

**Comment by supaiku-o** [comment]
I knew it's something trivial. Would be cool to have it at least noted somewhere in the docs.
The issue can be closed. Thanks guys!

**Comment by akien-mga** [comment]
This could be mentioned in the docs for `RegEx.compile()`, and/or a usage example could be added.

**Comment by leezh** [comment]
Yup, I'll make a pull request when I get home tonight.

**Comment by bojidar-bg** [comment]
Closed by #5846 but not done by GH automatically..
